executor/executor.go

```diff
@@ -21,7 +23,10 @@ func (e ErrStatus) Error() string {
 }
 
 // Run executes a slice of CommandDefs
-func Run(output io.Writer, job screwdriver.JobDef) error {
+func Run(path string, output io.Writer, job screwdriver.JobDef) error {
+	if err := chdir(path); err != nil {
```

Comment:
Yes, they would have to chdir each step. What do we want to happen here? We should decide if we want it to act like a single shell for all steps or independent and consistent environments.

What I don't like is the introduction of a side-effect by the executor. Calling a helper function in a package shouldn't result in the caller having a different working directory.

It also seems awkward to have some shell side-effects but not others. If we let the directory change in a step and carry over, we should probably try to propagate the environment as well - like current Screwdriver.